app/src/main/kotlin/ooo/akito/webmon/ui/createentry/CreateEntryActivity.kt

```diff
@@ -406,6 +414,86 @@ class CreateEntryActivity : AppCompatActivity() {
     }
   }
 
+  val CUSTOM_PREF_NAME = "Entry_data"
+
+  object PreferenceHelper {
+
+    val NAME = "NAME"
+    val URL = "URL"
+    val isDNSChecked = "DNSCheck"
+    val isLaissezFaireChecked = "LaissezCheck"
+
+    fun defaultPreference(context: Context): SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+
+    fun customPreference(context: Context, name: String): SharedPreferences = context.getSharedPreferences(name, Context.MODE_PRIVATE)
+
+    inline fun SharedPreferences.editMe(operation: (SharedPreferences.Editor) -> Unit) {
+      val editMe = edit()
+      operation(editMe)
+      editMe.apply()
+    }
+
+    var SharedPreferences.saveURL
+      get() = getString(URL, "")
+      set(value) {
+        editMe {
+          it.putString(URL, value)
+        }
+      }
+
+    var SharedPreferences.saveIsDNSChecked
+      get() = getBoolean(isDNSChecked, false)
+      set(value) {
+        editMe {
+          it.putBoolean(isDNSChecked, value)
+        }
+      }
+
+    var SharedPreferences.saveName
+      get() = getString(NAME, "")
+      set(value) {
+        editMe {
+          it.putString(NAME, value)
+        }
+      }
+
+    var SharedPreferences.saveIsLaissezFaireChecked
+      get() = getBoolean(isLaissezFaireChecked, false)
+      set(value) {
+        editMe {
+          it.putBoolean(isLaissezFaireChecked, value)
+        }
+      }
+  }
+
+  private var isSubmit_gen = false
+
+  override fun onPause() {
+    super.onPause()
+    val sp_gen = customPreference(this, CUSTOM_PREF_NAME)
+    if(isSubmit_gen){
+      sp_gen.saveName = ""
+      sp_gen.saveURL = ""
+      sp_gen.saveIsDNSChecked = false
+      sp_gen.saveIsLaissezFaireChecked = false
+    }else {
+      sp_gen.saveName = activityCreateEntryBinding.editName.text.toString()
+      sp_gen.saveURL = activityCreateEntryBinding.editUrl.text.toString()
+      sp_gen.saveIsDNSChecked = activityCreateEntryBinding.checkDNSRecords.isChecked
+      sp_gen.saveIsLaissezFaireChecked = activityCreateEntryBinding.isLaissezFaire.isChecked
```

Comment:
The TOR/Onion Setting is missing.

Reply:
I was unsure which aspect of the UI was related to this setting so I left it out. Can you send me a screenshot and I can add this?